### PR TITLE
Blockchain Transition Tests

### PR DESCRIPTION
**Pull request by winsvega**
replace TestNetwork  blockchain tests with Transition tests. 
the transition block now is 5 
the tests for transition are in folders 

bcEIP158ToMetropolis
bcFrontierToHomestead
bcHomesteadToDao
bcHomesteadToEIP150

The tests PR: https://github.com/ethereum/tests/pull/217

### CODE REVIEW

File: test/tools/jsontests/BlockChainTestsBoost.cpp [hunk 1]
```diff
@@ -53,15 +53,14 @@ class bcTestFixture {
...
-
 			return;
 		}
 		fillAllFilesInFolder(casename);
 	}
 
 	void fillAllFilesInFolder(string const& _folder)
 	{
-		std::string fillersPath = test::getTestPath() + "/src/BlockchainTestsFiller/" + _folder;
+		std::string fillersPath = test::getTestPath() + "/src/BlockchainTestsFiller/"  + _folder;
```
Comment (pirapira) [inline test/tools/jsontests/BlockChainTestsBoost.cpp]: ?

File: test/tools/jsontests/BlockChainTestsBoost.cpp [hunk 2]
```diff
@@ -78,7 +77,40 @@ class bcTestFixture {
 		for (auto const& file: files)
 		{
 			test::TestOutputHelper::setCurrentTestFileName(file.filename().string());
-			test::executeTests(file.filename().string(), "/BlockchainTests/"+_folder, "/BlockchainTestsFiller/"+_folder, dev::test::doBlockchainTestNoLog);
+			test::executeTests(file.filename().string(), "/BlockchainTests/"  + _folder, "/BlockchainTestsFiller/"  +_folder, dev::test::doBlockchainTestNoLog);
```
Comment (pirapira) [inline test/tools/jsontests/BlockChainTestsBoost.cpp]: One space is still missing `+_folder`.

File: test/tools/libtesteth/TestHelper.cpp [hunk 3]
```diff
@@ -103,29 +103,43 @@ string netIdToString(eth::Network _netId)
...
+		if (netIdToString(net) == _netname)
+			return net;
+
 	BOOST_ERROR(TestOutputHelper::testName() + " network not found: " + _netname);
 	return eth::Network::FrontierTest;
 }
 
 std::vector<eth::Network> const& getNetworks()
 {
+	//Networks for the test case execution when filling the tests
```
Comment (pirapira) [inline test/tools/libtesteth/TestHelper.cpp]: Why are the new networks not added here?

Comment (winsvega) [inline test/tools/libtesteth/TestHelper.cpp]: this are fork networks

File: libethashseal/GenesisInfo.cpp [hunk 4]
```diff
@@ -59,14 +77,15 @@ h256 const& dev::eth::genesisStateRoot(Network _n)
...
-	case Network::MetropolisTest: return c_genesisStateRootMetropolisTest;
+	case Network::Ropsten:
+	case Network::MainNetworkTest:
+	case Network::TransitionnetTest:
+	case Network::FrontierTest:
+	case Network::HomesteadTest: \
+	case Network::EIP150Test:
+	case Network::EIP158Test:
+	case Network::MetropolisTest:
+		return c_genesisDefaultStateRoot;
```
Comment (pirapira) [inline libethashseal/GenesisInfo.cpp]: Is this change related to the transition tests?

Comment (winsvega) [inline libethashseal/GenesisInfo.cpp]: it is more of optimization

File: libethashseal/genesis/test/EIP158ToMetropolisAt5Test.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "../../GenesisInfo.h"
+
+static std::string const c_genesisInfoEIP158ToMetropolisAt5Test = std::string() +
```
Comment (pirapira) [inline libethashseal/genesis/test/EIP158ToMetropolisAt5Test.cpp]: Why is this `std::string +` necessary?

File: libethashseal/genesis/test/homesteadToDaoAt5Test.cpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "../../GenesisInfo.h"
+
+static std::string const c_genesisInfoHomesteadToDaoAt5Test = std::string() +
```
Comment (pirapira) [inline libethashseal/genesis/test/homesteadToDaoAt5Test.cpp]: Again I see `std::string() + `.

File: libethashseal/genesis/test/EIP158ToMetropolisAt5Test.cpp (hunk 5 above)
Comment (winsvega) [inline libethashseal/genesis/test/EIP158ToMetropolisAt5Test.cpp]: this could be replaced with  '\'
